src/qml/filters/fisheye/ui.qml

```diff
@@ -0,0 +1,655 @@
+/*
+ * Copyright (c) 2022 Meltytech, LLC
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.12
+import QtQuick.Controls 2.12
+import QtQuick.Layouts 1.12
+import Shotcut.Controls 1.0 as Shotcut
+import QtQuick.Controls.Styles 1.1
+
+Item {
+    width: 300
+    height: 150
+    SystemPalette { id: activePalette }
+
+    //plugin input index values
+    property string focalRatioParam: '0'
+    property string deFishParam: '1'
+    property string lensTypeParam: '2'
+    property string scaePresetParam: '3'
+    property string scaleManualParam: '4'
+    property string qualityParam: '5'
+    property string aspectPresetParam: '6'
+    property string aspectManualParam: '7'
+
+    //default plugin values
+    property double focalRatioDefault: 0.5  //default was 0.0
+    property bool   deFishDefault: false
+    property double lensTypeDefault: 0.6    //default #3
+    property double scalePresetDefault: 0.6 //default #3
+    property double scaleManualDefault: 0.5
+    property double qualityDefault: 0.16
+    property double aspectPresetDefault: 0.0
+    property double aspectManualDefault: 0.5
+
+    //hide user sliders
+    property bool scaleShowSlider: false
+    property bool aspectShowSlider: false
+    property bool cameraShowNew: false    
+
+    //combobox presets text index
+    property int iDX_CAMERA: 8
+    property int iDX_WIDE: 9
+    property int iDX_RESULT: 10
+
+    property var defaultParameters: [focalRatioParam,deFishParam,lensTypeParam,scaePresetParam,
+                scaleManualParam,qualityParam,aspectPresetParam,aspectManualParam]
+
+
+    //preset data. used for comboboxes.
+    function getPresetData(index) {
+        switch (index) {
+            //Action
+            // tries to keep all 4 borders in frame. straighten left/right edge. zoomed center, makes motion look faster
+            //
+            //Focused
+            // close match gopro studio, trimed left/right edges, centre zoom slightly, makes motion look faster
+            //
+            //Linear
+            // minimal distortion at frame centre. edges have increased croped to suit
+
+
+            //HERO3
+            case  0: return [0.736, true, lensValue(4), scaleValue(1), 0.500, qualityValue(2), aspectValue(1), 0.500, 'HERO-3 1080', 'Wide', 'focused'] //todo: language?

```

Comment:
Change all of the "HERO-3" to simply "HERO3" and so on.
After we decide about presets make words like "Wide" translatable with `qsTr()`.
Change "focused" to "Focus" to be consistent?